### DISCUSSION TITLE
Using Clerk as IdP in AWS Cognito Identity Pool

### DISCUSSION THREAD

**Discussion by freivariable** [clerk/javascript#3587] [Support request - DO NOT OPEN HERE]
I am trying to use Clerk issued JWT tokens to acquire aws credentials using aws cognito identity client. I have configured the identity provider in aws cognito and configured the audience and issuer urls that match with the jwt tokens but AWS is still unable to verify the token. 
Here is the error I'm receiving when calling get-id operation on identity client: 

```
Invalid login token. Couldn't verify signed token.
```

**Comment by domjoseph** [comment]
Any luck with this?

**Comment by mishraditya05** [comment]
> I am trying to use Clerk issued JWT tokens to acquire aws credentials using aws cognito identity client. I have configured the identity provider in aws cognito and configured the audience and issuer urls that match with the jwt tokens but AWS is still unable to verify the token. Here is the error I'm receiving when calling get-id operation on identity client:
> 
> ```
> Invalid login token. Couldn't verify signed token.
> ```

How did you set this up, can you provide some reference ?

**Comment by mishraditya05** [comment]
Also does clerk allow us to access the OAuth ID Token it gets from OIDC Providers, which can be exchanged for AWS Credentials in Cognito IdentityPool.

**Comment by jescalan** [comment]
https://github.com/orgs/clerk/discussions/3829
